File: src/main/java/dev/slimevr/vr/trackers/TrackerFilters.java
package dev.slimevr.vr.trackers;

import java.util.HashMap;
import java.util.Map;


public enum TrackerFilters {

	NONE(0, "none"),
	SMOOTHING(1, "smoothing"),
	PREDICTION(2, "prediction");

	private static final Map<String, TrackerFilters> byConfigkey = new HashMap<>();

	static {
		for (TrackerFilters configVal : values()) {
			byConfigkey.put(configVal.configKey.toLowerCase(), configVal);
		}
	}

	public static final TrackerFilters[] values = values();
	public final int id;
	public final String configKey;

	TrackerFilters(int id, String configKey) {
		this.id = id;
		this.configKey = configKey;
	}

	public static TrackerFilters fromId(int id) {
		for (TrackerFilters filter : values) {
			if (filter.id == id)
				return filter;
		}
		return null;
	}

	public static TrackerFilters getByConfigkey(String configKey) {
		return configKey == null ? null : byConfigkey.get(configKey.toLowerCase());
	}
}

File: server/core/src/main/java/dev/slimevr/filtering/TrackerFilters.kt
package dev.slimevr.filtering

import java.util.*

enum class TrackerFilters(val id: UByte, val configKey: String) {
	NONE(0u, "none"),
	SMOOTHING(1u, "smoothing"),
	PREDICTION(2u, "prediction"),
	;

	companion object {
		private val byConfigkey: MutableMap<String, TrackerFilters> = HashMap()

		init {
			for (configVal in values()) {
				byConfigkey[configVal.configKey.lowercase(Locale.getDefault())] =
					configVal
			}
		}

		val values = values()

		@JvmStatic
		fun fromId(id: UByte): TrackerFilters? {
			for (filter in values) {
				if (filter.id == id) return filter
			}
			return null
		}

		@JvmStatic
		fun getByConfigkey(configKey: String?): TrackerFilters? = if (configKey == null) null else byConfigkey[configKey.lowercase(Locale.getDefault())]
	}
}
